set.mm

```diff
@@ -777824,6 +777899,211 @@ mean the category of preordered sets (classes).  However, "ProsetToCat"
     $}
   $}
 
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Monoids as categories
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c MndToCat $.
+
+  $( Class function defining monoids as categories. $)
+  cmndtc $a class MndToCat $.
+
+  ${
+    $d B x $.  $d M m x $.  $d m ph $.
+    $( Definition of the function converting a monoid to a category.  Example
+       3.3(4.e) of [Adamek] p. 24.
+
+       The definition of the base set is arbitrary.  The whole extensible
+       structure becomes the object here (see ~ mndtcbasval ) , instead of just
+       the base set, as is the case in Example 3.3(4.e) of [Adamek] p. 24.
+
+       The resulting category is defined entirely, up to isomorphism, by
+       ~ mndtcbas , ~ mndtchom , ~ mndtcco .  Use those instead.
+
+       See example 3.26(3) of [Adamek] p. 33 for more on isomorphism.
+
+       "MndToCat" was taken instead of "MndCat" because the latter might mean
+       the category of monoids.  (Contributed by Zhi Wang, 22-Sep-2024.)
+       (New usage is discouraged.) $)
+    df-mndtc $a |- MndToCat = ( m e. Mnd |-> { <. ( Base ` ndx ) , { m } >. ,
+          <. ( Hom ` ndx ) , { <. m , m , ( Base ` m ) >. } >. ,
+          <. ( comp ` ndx ) , { <. <. m , m , m >. , ( +g ` m ) >. } >. } ) $.
+
+    mndtcbas.c $e |- ( ph -> C = ( MndToCat ` M ) ) $.
+    mndtcbas.m $e |- ( ph -> M e. Mnd ) $.
+    $( Value of the category.  (Contributed by Zhi Wang, 22-Sep-2024.)
+       (New usage is discouraged.) $)
+    mndtcval $p |- ( ph -> C = { <. ( Base ` ndx ) , { M } >. ,
+          <. ( Hom ` ndx ) , { <. M , M , ( Base ` M ) >. } >. ,
+          <. ( comp ` ndx ) , { <. <. M , M , M >. , ( +g ` M ) >. } >. } ) $=
+      ( vm cmndtc cfv cnx cbs csn cop cotp cplusg ctp cmnd wceq opeq2d oteq123d
+      fveq2 chom cco wcel cv sneq id sneqd opeq12d tpeq123d df-mndtc tpex fvmpt
+      syl eqtrd ) ABCGHZIJHZCKZLZIUAHZCCCJHZMZKZLZIUBHZCCCMZCNHZLZKZLZOZDACPUCU
+      OVJQEFCUPFUDZKZLZUSVKVKVKJHZMZKZLZVDVKVKVKMZVKNHZLZKZLZOVJPGVKCQZVMURVQVC
+      WBVIWCVLUQUPVKCUERWCVPVBUSWCVOVAWCVKCVKCVNUTWCUFZWDVKCJTSUGRWCWAVHVDWCVTV
+      GWCVRVEVSVFWCVKCVKCVKCWDWDWDSVKCNTUHUGRUIFUJURVCVIUKULUMUN $.
+
+    mndtcbas.b $e |- ( ph -> B = ( Base ` C ) ) $.
+    $( The base set of the category.  (Contributed by Zhi Wang, 22-Sep-2024.)
+       (New usage is discouraged.) $)
+    mndtcbasval $p |- ( ph -> B = { M } ) $=
+      ( cbs cfv cnx csn cop chom cotp cco cplusg ctp mndtcval cvv c1 fveq2d cdc
+      wcel wceq snex c5 catstr baseid snsstp1 strfv mp1i 3eqtr4d ) ACHIJHIDKZLZ
+      JMIDDDHINKZLZJOIDDDNDPILKZLZQZHIZBUMACUSHACDEFRUAGUMSUCUMUTUDADUEUMUSHSTT
+      UFUBLUQUMUOUGUHUNUPURUIUJUKUL $.
```

Comment:
Maybe "_The base set of the category built from a monoid_" ?
This way, the comment would be self-contained.
(same remark for the other theorems)

Reply:
https://github.com/metamath/set.mm/pull/4242/commits/7d71ac3e296982cf20d346f748e616c1fd762028